api/provider/aws/test_aws/service_create_test.go

```diff
@@ -0,0 +1,211 @@
+	"github.com/quintilesims/layer0/api/tag"
+	awsc "github.com/quintilesims/layer0/common/aws"
+	"github.com/quintilesims/layer0/common/config/mock_config"
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestServiceCreate(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockAWS := awsc.NewMockClient(ctrl)
+	tagStore := tag.NewMemoryStore()
+	mockConfig := mock_config.NewMockAPIConfig(ctrl)
+
+	mockConfig.EXPECT().Instance().Return("test").AnyTimes()
+
+	tags := models.Tags{
+		{
+			EntityID:   "dpl_id",
+			EntityType: "deploy",
+			Key:        "arn",
+			Value:      "dpl_arn",
+		},
+	}
+
+	for _, tag := range tags {
+		if err := tagStore.Insert(tag); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	defer provider.SetEntityIDGenerator("svc_id")()
+
+	loadBalancerInput := &elb.DescribeLoadBalancersInput{}
+	loadBalancerInput.SetLoadBalancerNames([]*string{aws.String("l0-test-lb_id")})
+	loadBalancerInput.SetPageSize(1)
+
+	listener := &elb.Listener{}
+	listener.SetInstancePort(80)
+
+	listenerDescription := &elb.ListenerDescription{}
+	listenerDescription.SetListener(listener)
+
+	listenerDescriptions := []*elb.ListenerDescription{
+		listenerDescription,
+	}
+
+	loadBalancerDescription := &elb.LoadBalancerDescription{}
+	loadBalancerDescription.SetListenerDescriptions(listenerDescriptions)
+	loadBalancerDescription.SetLoadBalancerName("l0-test-lb_id")
+
+	loadBalancerDescriptions := []*elb.LoadBalancerDescription{
+		loadBalancerDescription,
+	}
+
+	loadBalancerOutput := &elb.DescribeLoadBalancersOutput{}
+	loadBalancerOutput.SetLoadBalancerDescriptions(loadBalancerDescriptions)
+
+	mockAWS.ELB.EXPECT().
+		DescribeLoadBalancers(loadBalancerInput).
+		Return(loadBalancerOutput, nil)
+
+	taskDefinitionInput := &ecs.DescribeTaskDefinitionInput{}
+	taskDefinitionInput.SetTaskDefinition("dpl_arn")
+
+	portMapping := &ecs.PortMapping{}
+	portMapping.SetContainerPort(int64(80))
+
+	portMappings := []*ecs.PortMapping{
+		portMapping,
+	}
+
+	containerDefinition := &ecs.ContainerDefinition{}
+	containerDefinition.SetName("ctn_name")
+	containerDefinition.SetPortMappings(portMappings)
+
+	containerDefinitions := []*ecs.ContainerDefinition{
+		containerDefinition,
+	}
+
+	taskDefinition := &ecs.TaskDefinition{
+		ContainerDefinitions: containerDefinitions,
+	}
+
+	taskDefinitionOutput := &ecs.DescribeTaskDefinitionOutput{}
+	taskDefinitionOutput.SetTaskDefinition(taskDefinition)
+
+	mockAWS.ECS.EXPECT().
+		DescribeTaskDefinition(taskDefinitionInput).
+		Return(taskDefinitionOutput, nil)
+
+	createServiceInput := &ecs.CreateServiceInput{}
+	createServiceInput.SetCluster("l0-test-env_id")
+	createServiceInput.SetDesiredCount(1)
+	createServiceInput.SetServiceName("l0-test-svc_id")
+	createServiceInput.SetTaskDefinition("dpl_arn")
+
+	loadBalancer := &ecs.LoadBalancer{}
+	loadBalancer.SetContainerName("ctn_name")
+	loadBalancer.SetContainerPort(80)
```

Comment:
Ditto with the default `provider.DefaultLoadBalancerPort.ContainerPort`

Reply:
Actually, disregard this. We're not talking about a LoadBalancer here, so that would be a little awkward to mix and match the default. Perhaps a new default could be defined?

Reply:
Here the default load balancer port is more appropriate since we are actually dealing with a load balancer object here.

Reply:
✅